### DISCUSSION TITLE
Allow pasting / uploading QR codes as an image, as an alternative to using the built-in capture tool

### DISCUSSION THREAD

**Discussion by mzrnsh** [Authenticator-Extension/Authenticator#1108] [Ideas]
On Firefox when I try to add a QR code it requests the following permissions:

![CleanShot 2023-09-19 at 12 06 08](https://github.com/Authenticator-Extension/Authenticator/assets/10742145/f1c3b9d6-624a-4232-993f-b74f466b0d44)

I don't know if I am being too paranoic here, but I am very hesitant to give such permissions to anyone. So instead, I'd prefer to use Mac's region capture tool (CMD+SHIFT+4) and paste the QR code image, or if that's too much, store it as an image locally, and then upload as a file.

**Comment by Sneezry** [comment]
For the permission issue, please refer to https://github.com/Authenticator-Extension/Authenticator/issues/392#issuecomment-541931294

You can upload QR codes as an image. If you have a screenshot of a QR image, use Import QR Image Backup. You can select multiple images to import more than one account at a time. See https://authenticator.cc/docs/en/export-and-import

**Reply by mzrnsh** [reply]
Thanks, I would have never guessed this option would be hidden away in backup import feature. Will try it the next time. My "workaround" was way uglier:

- Install on Chrome
- Use QR capture there
- Download backup as txt
- Add new account in FF manually
- Use values extracted from the txt
- 🤔
